pandas/io/formats/csvs.py

```diff
@@ -157,13 +171,18 @@ def save(self):
             self._save()
 
         finally:
-            # GH 17778 handles zip compression for byte strings separately.
-            buf = f.getvalue()
-            if path_or_buf:
-                f, handles = _get_handle(path_or_buf, self.mode,
-                                         encoding=encoding,
-                                         compression=self.compression)
-                f.write(buf)
+            if is_zip:
+                # GH 17778 handles zip compression separately.
+                buf = f.getvalue()
+                try:
+                    self.path_or_buf.write(buf)
+                except AttributeError:
+                    f, handles = _get_handle(self.path_or_buf, self.mode,
```

Comment:
can you add a comment on when the except happens?

Reply:
done.